loader/0002-quilt.mod.json.md

```diff
@@ -0,0 +1,362 @@
+| String | True     |
+
+Must conform to the [Semantic Versioning 2.0.0 specification](https://semver.org/). In a development environment, the value `${version}` can be used as a placeholder by quilt-gradle to be replaced on building the resulting JAR.
+
+### The `entrypoints` field
+| Type   | Required |
+|--------|----------|
+| Object | False    |
+
+A collection of `key: value` pairs, where each key is the type of the entrypoints specified and each values is either a single entrypoint or an array of entrypoints. An entrypoint is an object with the following keys:
+* adapter — Language adapter to use for this entrypoint
+* value — Points to an implementation of the entrypoint. For the default JVM language adapter, this can be in either of the following forms:
+    * `my.package.MyClass` — A class to be instantiated and used
+    * `my.package.MyClass::thing` — A static field containing an instance of the entrypoint or a method handle for entrypoints that are functional interfaces
+
+If an entrypoint does not need to specify a language adapter other than the default language adapter, the entrypoint can be represented simply as the value string instead.
+
+
+### The `plugins` field
+| Type   | Required |
+|--------|----------|
+| Array  | False    |
+
+An array of loader plugins. A plugin is an object with the following keys:
+* adapter — Language adapter to use for this plugin
+* value — Points to an implementation of the `LoaderPlugin` interface. Can be in either of the following forms:
+    * `my.package.MyClass` — A class to be instantiated and used
+    * `my.package.MyClass::thing` — A static field containing an instance of a `LoaderPlugin`
+
+If a plugin does not need to specify a language adapter other than the default language adapter, the plugin can be represented simply as the value string instead.
+
+### The `jars` field
+| Type   | Required |
+|--------|----------|
+| Array  | False    |
+
+A list of paths to nested JAR files to load, relative to the root directory inside of the mods JAR.
+
+### The `language_adapters` field
+| Type   | Required |
+|--------|----------|
+| Object | False    |
+
+A collection of `key: value` pairs, where each key is the namespace of a language adapter and the value is an implementation of the `LanguageAdapter` interface.
+
+### The `depends` field
+| Type   | Required |
+|--------|----------|
+| Object | False    |
+
+Defines mods that this mod will not function without.
+
+A collection of `key: value` pairs, where each key is in the form of either `mavenGroup:modId` or `modId` and each value is either a [dependency object]() or an array of dependency objects. If an array of dependency objects is provided, the dependency matches if it matches ANY of the dependency objects.
+
+#### Dependency Object
+A dependency object is made up of a version specifier and a reason for that dependency. It may be represented in any of the following ways:
+```json5
+// As a JSON object, with both available fields
```

Comment:
Making a mod that interfaces with as many others as it can, the lack of something like this is one of the few things that have bugged me about `fabric.mod.json`s.

Reply:
It might also be useful to add a "side" option, sometimes dependencies arn't needed on the 'server' side. the only option with the current stuff is to not add the mod to my depends list.

Reply:
@wagyourtail I think that's part of the discussion above? (although swap `side` for `environment`) https://github.com/QuiltMC/rfcs/pull/2#discussion_r590935371